doc/DOCUMENTATION.org

```diff
@@ -2338,7 +2338,7 @@ Windows manipulation commands (start with ~w~):
 | ~SPC w c C~            | toggle visual distraction free mode                                         |
 | ~SPC w c .~            | center buffer and enable centering transient state                          |
 | ~SPC w d~              | delete a window                                                             |
-| ~SPC u SPC w d~        | delete a window and its current buffer (does not delete the file)           |
```

Comment:
I would like to keep the old behaviour, users should have a choice between the `spc u` way and the pure `num` flavor.

Reply:
There already exist two bindings for deleting a window and its buffer `SPC w x` and `SPC b x`. Therefore, I argued the binding you are referring to, is somewhat redundant.

This PR suggests that we can make better use of the `SPC w d` prefix, namely, to use it for quickly deleting other windows.
Currently, the only way to quickly delete another window is via`ace-delete-window` for which there is no default binding. However, I would argue that using a numeric prefix with `SPC w d` is a better alternative to `ace-delete-window`, as ace shows labels only after invoking it, while a window number is known already before invoking `SPC w d`. Furthermore, unlike `ace-delete-window`, this alternative does not require creating some extra (custom) binding, while also, it does not depend on ace.

Reply:
I think adding this numerical prefix support is a good idea and would be happy to see it merged. Though keeping the old <kbd>SPC u</kbd>-prefix behaviour would also be preferable to me.

Hence I have set up a draft PR implementing this including a few changes. It handles some edge cases like entering a non-existing window number (which should not delete the current window instead) or having the package winum excluded.
Also do we actually need to reimplement the `kill-buffer-and-window` function? Wrapping it in a `(with-selected-window ...)` form seems safe to me.
@dalanicolai Would you mind having a quick look at #16425? I have added you as a co-author if that's fine with you.

> Currently, the only way to quickly delete another window is via `ace-delete-window`

There exists <kbd>SPC u SPC NUM</kbd> as an alternative, but I still think it would be nice to support <kbd>NUM SPC w d</kbd>, especially in combination with the proposed <kbd>NUM SPC w x</kbd>.